### PR TITLE
Add methods to get results by query id

### PR DESCRIPTION
**Pull request by TheEdgeOfRage**
Considering that we also have a CLI implementation that is pretty bad atm (you control the action by either setting -e or -q), I think I wanna do a rework of the CLI. I don't really want to make a breaking change just like that, but I don't know what the correct protocol is for go? Is it enough to tag the current main commit (before this PR) as v1 and then create a new v2 tag to allow people to keep using v1 if they want?

### CODE REVIEW

File: dune/dune.go [hunk 1]
```diff
@@ -167,7 +173,45 @@ func (c *duneClient) QueryResults(executionID string) (*models.ResultsResponse,
...
+
+	// we read whole result into ram here. if there was a paginated API we wouldn't need to
+	var buf bytes.Buffer
+	defer resp.Body.Close()
+	_, err = buf.ReadFrom(resp.Body)
+	return &buf, err
+}
+
+func (c *duneClient) QueryResultsByQueryID(queryID string) (*models.ResultsResponse, error) {
+	resultsURL := fmt.Sprintf(queryResultsURLTemplate, c.env.Host, queryID)
```
Comment (diegoximenes) [inline dune/dune.go:194]: That is the only line that is different from QueryResults func.
How about creating a function `func (c *duneClient) getQueryResults(url string) (*models.ResultsResponse, error)`, and call it from QueryResultsByQueryID and QueryResults?

The same comment is valid for QueryResultsCSV and QueryResultsCSVByQueryID.

**Comment by diegoximenes** [comment]
Maybe this can help you https://go.dev/doc/modules/major-version.

But why this CLI needs a breaking change?

And why this change in the CLI is a priority at the moment?
I am asking this because I don't have visibility of how many users this CLI has, and how those users update their CLI.

File: dune/dune.go (hunk 1 above)
Comment (TheEdgeOfRage) [reply dune/dune.go:194]: yeah, fair :smile:

**Comment by TheEdgeOfRage** [comment]
The CLI change is not a priority really, but if I'm gonna make a breaking change, I would like to do it as early as possible, to impact the least amount of users. As to why, it's not technically necessary, but I didn't think it through when I started developing this library and as we start adding more functionality to the client, is will become very difficult to maintain and add new functionality to the CLI. I would like to switch to a CLI framework, like cobra

**Review by diegoximenes — APPROVED** [review]

**Pull request merged**